### DISCUSSION TITLE
Attach is slow

### DISCUSSION THREAD

**Discussion by gaz77a** [getgauge/taiko#2521] [General]
When I run `attach` it takes 22s to execute.
If I use the browser manually the attach is instant.
I have the test ensure the field exists before using attach and I do a console log before and after the attach.
The size of the file is 80Kb and is attached successfully after the 22 seconds.

Any thoughts?

The code being run is similar to
`await attach("/someDir/someFile.txt", fileField({ id: "fileInput" }), { force: true });`

Thanks in advance

**Comment by NivedhaSenthil** [comment]
could be because the action is waiting for some navigation to complete.. if you dont want to wait for the resultant actions like XHRs you could pass `{waitForNavigation: false}` option

**Reply by gaz77a** [reply]
Thanks for the quick reply.
I have tried your suggestion however there is no change.
I also couldn't see `waitForNavigation` as an option in the taiko documentation for attach.

Here is the code I tried 
`  await attach("/someDir/someFile.txt", fileField({ id: "fileInput" }), {
    force: true,
    waitForNavigation: false,
  });`

Thanks in advance

**Reply by NivedhaSenthil** [reply]
Thanks for checking on this, I just realised we don't wait for navigation during attach. Can you please provide a sample where you are seeing this lag for us understand this scenario better ? As the sample I tried seems to be working fine without any lag,

```
const { openBrowser, goto, fileField, attach, closeBrowser } = require('taiko');
(async () => {
    try {
        await openBrowser();
        await goto("https://www.w3schools.com/howto/tryit.asp?filename=tryhow_html_file_upload_button");
        await attach("/../Downloads/map.jpeg",fileField());
    } catch (error) {
        console.error(error);
    } finally {
        await closeBrowser();
    }
})();
```

**Reply by gaz77a** [reply]
Thanks @NivedhaSenthil 
Here is the actual code.
I have placed a console log before and after to narrow down that it was the attach that was slow.
The first console log shows and then the second one takes 22s.

```
  console.log("fieldLabel", await fileField({ id: "fileInput" }).exists());
  await attach(filePath, fileField({ id: "fileInput" }), { force: true  });
  console.log("after attach");
```

**Reply by NivedhaSenthil** [reply]
Can you please also share a snippet of html and js where we can replicate this ?

**Reply by gaz77a** [reply]
If I extract the code for the modal window so I can share it with you I can't reproduce the issue.
The issue is only when using taiko with our full application.
Is there any tracing I can view locally to see what Taiko is doing during the 22s?

Thanks in advance

**Reply by zabil** [reply]
There's info about enabling logs at https://docs.taiko.dev/frequently_asked_questions/#how-do-i-enable-debug-log-events%3F
